src/Parser.jl

```diff
@@ -364,11 +370,21 @@ function number_from_bytes(pc::ParserContext,
 end
 
 
-function parse_number(pc::ParserContext, ps::ParserState)
+function parse_number(pc::ParserContext{<:Any,<:Any,AllowNanInf}, ps::ParserState) where AllowNanInf
     # Determine the end of the floating point by skipping past ASCII values
     # 0-9, +, -, e, E, and .
     number = ps.utf8array
     isint = true
+    negative = false
+
+    c = current(ps)
+
+    # Parse and keep track of initial minus sign (for parsing -Infinity)
+    if AllowNanInf && c == MINUS_SIGN
+        push!(number, UInt8(c)) # save in case the next character is a number
+        negative = true
+        incr!(ps)
```

Comment:
same comment here as below; does this still preserve correct behaviour with regular negative numbers?

Reply:
This one had issues.  Parsing negative numbers is okay, but I forgot to reset the parse buffer, so the `-` sign was still in the buffer at the start of the next parse.

The current test catches this, but I'll add a more specific one.

Reply:
Tests have been updated.